pkg/agent/util/winnet/net_windows.go

```diff
@@ -0,0 +1,830 @@
+	if err != nil {
+		return fmt.Errorf("unable to list Windows IPForward rows: %v", err)
+	}
+	for i := range rows {
+		row := rows[i]
+		if row.DestinationPrefix.EqualsTo(route.DestinationSubnet) && row.Index == uint32(route.LinkIndex) && row.NextHop.IP().Equal(route.GatewayAddress) {
+			if err := antreaNetIO.DeleteIPForwardEntry(&row); err != nil {
+				return fmt.Errorf("failed to delete existing route with nextHop %s: %v", route.GatewayAddress, err)
+			}
+		}
+	}
+	return nil
+}
+
+func (h *Handle) ReplaceNetRoute(route *Route) error {
+	if route == nil || route.DestinationSubnet == nil {
+		return nil
+	}
+	family := AddressFamilyByIP(route.DestinationSubnet.IP)
+	rows, err := antreaNetIO.ListIPForwardRows(family)
+	if err != nil {
+		return fmt.Errorf("unable to list Windows IPForward rows: %v", err)
+	}
+	for i := range rows {
+		row := rows[i]
+		if row.DestinationPrefix.EqualsTo(route.DestinationSubnet) && row.Index == uint32(route.LinkIndex) {
+			if row.NextHop.IP().Equal(route.GatewayAddress) {
+				return nil
+			} else {
+				if err := antreaNetIO.DeleteIPForwardEntry(&row); err != nil {
+					return fmt.Errorf("failed to delete existing route with nextHop %s: %v", route.GatewayAddress, err)
+				}
+			}
+		}
+	}
+	return h.AddNetRoute(route)
+}
+
+func (h *Handle) RouteListFiltered(family uint16, filter *Route, filterMask uint64) ([]Route, error) {
+	rows, err := antreaNetIO.ListIPForwardRows(family)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list Windows IPForward rows: %v", err)
+	}
+	rts := make([]Route, 0, len(rows))
+	for i := range rows {
+		route := routeFromIPForwardRow(&rows[i])
+		if filter != nil {
+			if filterMask&RT_FILTER_IF != 0 && filter.LinkIndex != route.LinkIndex {
+				continue
+			}
+			if filterMask&RT_FILTER_DST != 0 && !iputil.IPNetEqual(filter.DestinationSubnet, route.DestinationSubnet) {
+				continue
+			}
+			if filterMask&RT_FILTER_GW != 0 && !filter.GatewayAddress.Equal(route.GatewayAddress) {
+				continue
+			}
+			if filterMask&RT_FILTER_METRIC != 0 && filter.RouteMetric != route.RouteMetric {
+				continue
+			}
+		}
+		rts = append(rts, *route)
+	}
+	return rts, nil
+}
+
+func parseCmdResult(result string, columns int) [][]string {
+	scanner := bufio.NewScanner(strings.NewReader(result))
+	parsed := [][]string{}
+	for scanner.Scan() {
+		items := strings.Fields(scanner.Text())
+		if len(items) < columns {
+			// Skip if an empty line or something similar
+			continue
+		}
+		parsed = append(parsed, items)
+	}
+	return parsed
+}
+
+func (h *Handle) AddNetNat(netNatName string, subnetCIDR *net.IPNet) error {
+	cmd := fmt.Sprintf("Get-NetNat -Name %s | Select-Object InternalIPInterfaceAddressPrefix | Format-Table -HideTableHeaders", netNatName)
+	if internalNet, err := runCommand(cmd); err != nil {
+		if !strings.Contains(err.Error(), "No MSFT_NetNat objects found") {
+			klog.ErrorS(err, "Failed to check the existing netnat", "name", netNatName)
+			return err
```

Comment:
Yes, it was copied from the original file. I'll check and refine such code.

Reply:
@wenyingd One more thing, do we have any special reasons for adding the log here before?

Reply:
@hongliangl I don't think this has been addressed (or at least I see other occurrences of `klog.Error` + `return err`). But since it was not introduced by you, we can still merge this PR.

Reply:
My bad, I didn't notice that there are some code like this in file `pkg/agent/util/net_windows.go` and only addressed file `pkg/agent/util/winnet/net_windows.go`